Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py

```diff
@@ -0,0 +1,632 @@
+
+        # fetch only incidents that weren't fetched in the last run and their severity is at least min_severity
+        if incident.get('ID') not in last_fetch_ids and incident_severity >= min_severity:
+            incident_created_time = dateparser.parse(incident.get('CreatedTimeUTC'))
+            incident = {
+                'name': '[Azure Sentinel] ' + incident.get('Title'),
+                'occurred': incident.get('CreatedTimeUTC'),
+                'severity': incident_severity,
+                'rawJSON': json.dumps(incident)
+            }
+
+            incidents.append(incident)
+            current_fetch_ids.append(incident.get('ID'))
+
+            # Update last run to the latest fetch time
+            if incident_created_time > latest_created_time:
+                latest_created_time = incident_created_time
+
+    next_run = {
+        'last_fetch_time': latest_created_time.strftime(DATE_FORMAT),
+        'last_fetch_ids': current_fetch_ids
+    }
+    return next_run, incidents
+
+
+def main():
+    """
+        PARSE AND VALIDATE INTEGRATION PARAMS
+    """
+    params = demisto.params()
+
+    LOG(f'Command being called is {demisto.command()}')
+    try:
+        client = Client(
+            self_deployed=params.get('self_deployed', False),
+            auth_and_token_url=params.get('auth_id', ''),
+            refresh_token=params.get('refresh_token', ''),
+            enc_key=params.get('enc_key', ''),
+            redirect_uri=params.get('redirect_uri', ''),
+            auth_code=params.get('auth_code', ''),
+            subscription_id=params.get('subscriptionID', ''),
+            resource_group_name=params.get('resourceGroupName', ''),
+            workspace_name=params.get('workspaceName', ''),
+            verify=not params.get('insecure', False),
+            proxy=params.get('proxy', False)
+        )
+
+        commands = {
+            'azure-sentinel-get-incident-by-id': get_incident_by_id_command,
+            'azure-sentinel-list-incidents': list_incidents_command,
+            'azure-sentinel-update-incident': update_incident_command,
+            'azure-sentinel-delete-incident': delete_incident_command,
+            'azure-sentinel-list-incident-comments': list_incident_comments_command,
+            'azure-sentinel-incident-add-comment': incident_add_comment_command,
+            'azure-sentinel-list-incident-relations': list_incident_relations_command,
+            'azure-sentinel-get-entity-by-id': get_entity_by_id_command,
+            'azure-sentinel-list-entity-relations': list_entity_relations_command
+        }
+
+        if demisto.command() == 'test-module':
+            # cannot use test module due to the lack of ability to set refresh token to integration context
+            raise Exception("Please use !azure-sentinel-test instead")
+
+        elif demisto.command() == 'azure-sentinel-test':
+            test_connection(client)
```

Comment:
why?

Reply:
Because it doesn't work otherwise. I saw that Ika did it too in mail listener